test/api.jl

```diff
@@ -115,5 +116,30 @@ recommendation = provide_recommendation(trial_vector)
 recommendation_multi = provide_recommendation(trial_vector_multihyperparams)
 @test recommendation_multi == Dict(:x => 3, :y => 7)
 
+# Test case reproducing the issue in “obs_memo” function from the higher level
+space = Dict(
+    :a => HP.Choice(:a, [8, 9, 10]),
+    :b => HP.Choice(:b, [1, 2, 3])
+    )
+
+Random.seed!(10)
+trials = TreeParzen.Trials.Trial[]
+
+for i in 1:30
+    trial = ask(space, trials, TreeParzen.Config())
+    println("Iteration: ", i, ", ", trial.hyperparams)
+    tell!(trials, trial, trial.hyperparams[:b]/trial.hyperparams[:a])
+end
+# With this example space and loss function, expected vals for the best param should be different
+# but with the issue in “obs_memo” function, they are equal due to same nid
+@test trials[30].vals[:a] == trials[30].vals[:b]
```

Comment:
What you should actually do here is the test result you expect, which is that the values don't match. Plus, if i understand correctly, we expect trials 1:25 to be valid samples (because its the random sampling regime) and we expect 26:30 to be invalid samples (values coming from the other distribution)

It would be better to actually write that test in fully.

Finally, knowing that this will "break the ci", you can use `@test_broken` macro to show you expect a broken test. When it gets fixed, the broken macro will start failing, and you can switch it back to `@test`

Reply:
Yes exactly we test the result we expect here. For this loss function, 
`trial.hyperparams[:b]/trial.hyperparams[:a]` we have the smallest loss when "a" is the highest which is 10 here and "b" is the lowest which is 1. Example trials[i].vals[:a] is not returning the value of the space, it returns their index. 
In this test I am showing the bug by testing that when the last iteration is returned for a is index 3, the same index is returned for b due to that bug in “obs_memo” function. There is no point to test what are in trials 1:25 as the issue in “obs_memo” function happens from 21 to end of iteration by returning the wrong nid `    nid = obs_memo(item, params)
` and the wrong posterior outputs